starlette/middleware/base.py

```diff
@@ -44,14 +46,20 @@ def __init__(self, app: ASGIApp, dispatch: DispatchFunction = None) -> None:
         if message is None:
             task.result()
             raise RuntimeError("No response returned.")
-        assert message["type"] == "http.response.start"
+
+        if "http.response.template" in scope.get("extensions", {}):
```

Comment:
This is always True (https://codecov.io/gh/blueyed/starlette/compare/ef49e73e1043fdf7ce409a190ab632cfd568953e...e3ec97fcb16de9f3bb645ddb67495349c124dbb2/diff#D4-50).
I.e. we should maybe test with a TestClient that does not use it?!